app/src/main/java/edu/wisc/ece/pinpoint/pages/profile/EditProfileFragment.java

```diff
@@ -202,4 +230,57 @@ private void launchLocationAutocomplete(View view, boolean isFocused) {
                             requestedFields).setTypesFilter(placesFilter).build(requireContext()));
         }
     }
+
+    private void takePicture() {
+
+        Intent pictureIntent = new Intent(
+                MediaStore.ACTION_IMAGE_CAPTURE).addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        if(pictureIntent.resolveActivity(this.getContext().getPackageManager()) != null){
+            //Create a file to store the image
+            try {
+                photoFile = createImageFile();
+            } catch (IOException ex) {
+                Toast toast = Toast.makeText(this.getContext(), getString(R.string.file_creation_failed_message), Toast.LENGTH_SHORT);
+                toast.show();
+                return;
+            }
+            if (photoFile != null) {
+                photo = FileProvider.getUriForFile(this.getContext(),
+                        BuildConfig.APPLICATION_ID + ".fileprovider", photoFile);
+                pictureIntent.putExtra(MediaStore.EXTRA_OUTPUT,
+                        photo);
+            }
+        }
+
+        // launch gallery opening intent
+        photoTakerLauncher.launch(pictureIntent);
+    }
+
+    private File createImageFile() throws IOException {
+        String timeStamp =
+                new SimpleDateFormat("yyyyMMdd_HHmmss",
+                        Locale.getDefault()).format(new Date());
+        String imageFileName = "IMG_" + timeStamp + "_";
+        File storageDir =
+                this.getContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+        File tmpFile = File.createTempFile(
+                imageFileName,  /* prefix */
+                ".jpg",         /* suffix */
+                storageDir      /* directory */
+        );
+
+        return tmpFile;
```

Comment:
You don't need to make a variable for this, just return `File.createTempFile(...)`